pkg/orchestrator/orchestrator.go

```diff
@@ -1249,6 +1285,131 @@ func (s *ControllerServer) resourceAttach(c *gin.Context, resourceInfo *Resource
 	c.JSON(http.StatusOK, attachResourceResp)
 }
 
+// Returns the resource info and cloud client for a given resource
+func (s *ControllerServer) getResourceInfo(c *gin.Context) (resourceInfo *ResourceInfo, cloudClient string, err error) {
+	resourceInfo, cloudClient, err = s.getAndValidateResourceURLParams(c, false)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if resourceInfo.name == "" {
+		return nil, "", errors.New("Resource name not specified")
+	}
+
+	// Get tag from tag service
+	tagName := getTagName(resourceInfo.namespace, resourceInfo.cloud, resourceInfo.name)
+	tag, err := s.getTagWithName(tagName)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Check if tag exists
+	if !isTagValid(tag) {
+		return nil, "", errors.New("Resource Tag does not exist")
+	}
+
+	resourceInfo.uri = *tag.Uri
+
+	return resourceInfo, cloudClient, nil
+}
+
+func (s *ControllerServer) checkResource(c *gin.Context) {
+	// Get resource info
+	resourceInfo, cloudClient, err := s.getResourceInfo(c)
+	if err != nil {
+		c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+		return
+	}
+
+	// Create connection to cloud plugin
+	conn, err := grpc.NewClient(cloudClient, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+		return
+	}
+	defer conn.Close()
+
+	// Send RPC to client to check resource
+	client := paragliderpb.NewCloudPluginClient(conn)
+	checkReq := &paragliderpb.CheckResourceRequest{Namespace: resourceInfo.namespace, Resource: resourceInfo.uri}
+	checkResp, err := client.CheckResource(context.Background(), checkReq)
+	if err != nil {
+		c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+		return
+	}
+	// Not checking if resource does not exist because if it doesn't,
+	// it would be the only error code in the Errors array and would be reported to client
+
+	// Create map of error codes for fast lookup
+	checkErrs := map[paragliderpb.ErrorCode]bool{}
+	for _, code := range checkResp.Errors {
+		checkErrs[code] = true
+	}
+
+	// Add both valid and invalid results to the map
+	// Error codes that don't exist(i.e valid messages) are identified by negative keys
+	errorMap := map[int32]string{}
```

Comment:
I think I am confused on why this additional logic is necessary. Why can't we just return the list of errors to the client?

Reply:
I've deleted this, and rewritten the implementation in the updated PR